### PR TITLE
[ELY-1255] Elytron XML certificate private-key parsing

### PR DESCRIPTION
**Pull request by honza889**
* added private-key parsing
* fixed pem element parsing (`hasText()` does not make sense before `getElementText()` call)
* dead code clearing (dont check end of document in end of authentication-client, as it is not root element already)

https://issues.jboss.org/browse/ELY-1255
https://issues.jboss.org/browse/JBEAP-11695

### CODE REVIEW

File: src/main/java/org/wildfly/security/pem/Pem.java [hunk 1]
```diff
@@ -48,8 +50,9 @@
  */
 public final class Pem {
     private static final Pattern VALID_LABEL = Pattern.compile("[^ -~&&[^-]]");
-    private static final String PUBLIC_KEY_FORMAT = "PUBLIC KEY";
+    private static final String PUBLIC_KEY_FORMAT = "PUBLICKEY";
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/pem/Pem.java]: PKCS#8 uses a space here, and PKCS#1 further gives e.g. "RSA PUBLIC KEY".  We need to support all the standard formats explicitly, and we also don't want to accept invalid input (e.g. "PUBLIC     KEY").

File: src/main/java/org/wildfly/security/pem/Pem.java [hunk 2]
```diff
@@ -67,22 +70,22 @@
     public static <R> R parsePemContent(CodePointIterator pemContent, BiFunction<String, ByteIterator, R> contentFunction) throws IllegalArgumentException {
         Assert.checkNotNullParam("pemContent", pemContent);
         Assert.checkNotNullParam("contentFunction", contentFunction);
-        pemContent = pemContent.skipCrLf();
-        if (! pemContent.limitedTo(11).contentEquals("-----BEGIN ")) {
+        pemContent = pemContent.skip(Character::isWhitespace).skipCrLf();
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/pem/Pem.java]: This in particular allows invalid input to be accepted.

**Comment by honza889** [comment]
[HALT] will improve checking to be less tolerant

**Comment by honza889** [comment]
@dmlloyd Improved to require exact PEM begin/end block. Is it ok now?

**Pull request merged**